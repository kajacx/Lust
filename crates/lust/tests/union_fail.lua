-- @type number | boolean | string
local empty = nil

-- @type boolean | number
local flag = true or "Hello"

-- @type string?
local value = 15

-- @type number? | boolean
local text = nil or "Hello world!"

-- @type number | string
local valueAgain = value or "text"
