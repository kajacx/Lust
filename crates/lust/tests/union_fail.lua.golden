tests/union_fail.lua:2:1: Cannot assign value 'TODO: Nil' of type 'Nil' into the variable 'empty' of type 'Union(UnionType { variants: [Number, True, False, String] })'.
tests/union_fail.lua:5:1: Cannot assign value 'TODO: OrOperation(OrOperation { left: BooleanLiteral(true), right: StringLiteral("Hello") })' of type 'Union(UnionType { variants: [True, String] })' into the variable 'flag' of type 'Union(UnionType { variants: [True, False, Number] })'.
tests/union_fail.lua:8:1: Cannot assign value 'TODO: NumberLiteral(15.0)' of type 'Number' into the variable 'value' of type 'Union(UnionType { variants: [String, Nil] })'.
tests/union_fail.lua:11:1: Cannot assign value 'TODO: OrOperation(OrOperation { left: Nil, right: StringLiteral("Hello world!") })' of type 'String' into the variable 'text' of type 'Union(UnionType { variants: [Number, Nil, True, False] })'.
tests/union_fail.lua:14:1: Cannot assign value 'TODO: OrOperation(OrOperation { left: VarName("flag"), right: StringLiteral("text") })' of type 'Union(UnionType { variants: [True, Number, String] })' into the variable 'valueAgain' of type 'Union(UnionType { variants: [Number, String] })'.
